modules/javafx.controls/src/main/java/javafx/scene/control/skin/ListCellSkin.java

```diff
@@ -127,7 +96,8 @@ private void setupListeners() {
 
     /** {@inheritDoc} */
     @Override protected double computePrefHeight(double width, double topInset, double rightInset, double bottomInset, double leftInset) {
-        if (fixedCellSizeEnabled) {
+        double fixedCellSize = getFixedCellSize();
+        if (fixedCellSize > 0) {
```

Comment:
These compute methods get invoked multiple times during each layout pass(10s of times). Fetching the fixed cell size on each call to these methods seems to be costly and repeated operation compared to previous boolean check. I think we should keep the previous way of handling it: registering the change listener to `listView.fixedCellSizeProperty()`.

Reply:
ehh .. last time I did such micro-optimization was in the 80ies of last century ;) 

Are there any performance measurements anywhere to demonstrate the impact?

Reply:
a bit less flippant: really interested in the measurements - certainly, they are somewhere but can't find anything. Any pointer where to look?

Reply:
Even I do not know if there are any performance measurements for such scenario. :) May be @kevinrushforth could help in that. 
I observed that for a ListView with only one item, these three methods combinely are called 97 times when the Stage is displayed for the first time and 45 times when the item is selected.

Reply:
hmm, do you mean a list with height of one cell only? 

My experiment (having counters per instance and per computeXXHeight), logs 1 (or 2) for computePref with fixed size and 5 (1 each for min/max and 3 for pref) without fixedSize per layout pass. So it's 5* access of either the local field or the listView method .. wouldn't expect any problems, but then assumed performance is reading in coffee ground :)

Let's hope there are some metrics to use :)

Reply:
> hmm, do you mean a list with height of one cell only?

Yes, List with one cell height. 
In continuation to my previous comment, looks like 8 of the first 23 calls when window is displayed are NOT for the one visible cell. These 8 calls are for some intermediate cell.
I added a single counter for all three `computeXXHeight` methods and verified with a list of height of one cell.
When `fixedCellSize` is not set, total combined number of calls for a cell are,
1:-  15, when Stage is displayed for the first time.(+8 for an intermediate cell, not sure why)
2:-  5, when the list item is selected.
3:-  5, when the Window is resized.

Reply:
something similar as I'm seeing - that's good :) Uploaded my [play code](https://gist.github.com/kleopatra/37ac1bffa0e2ad7580cd55344237cdca) to gist: the idea is to do something (like f.i. moving the selection), then press f1 to log the calls to each cell (the skin has a final instance counter and counters for calling the compute methods). 

As much fun as this is (really like to dig until I'm dirty all over :) - at the end of the day we'll need some measurements (doing these is not so much fun, still hoping something already available)

Reply:
Normally we implement this sort of lazy evaluation (caching + invalidation) when there is a computation that is being saved, or when a micro-benchmark shows that something is called 1000s of times (e.g., in the inner loop of some common operation).

I don't see either being the case here, but it's possible I missed something. The method in question, `getFixedCellSize()` just calls `ListView::getFixedCellSize` which just returns the value of a property. I suspect that any performance hit would be down in the noise.

So I think the complexity of the existing mechanism isn't justified. Removing the listener as the PR proposes to do, which is possible since the listener isn't being used to trigger an operation, seems like a win as well.

I am inclined to approve this fix even without performance measurements, since it seems unlikely they would show a problem.

@arapte if you want to measure it further before you approve it, that would be fine.

Reply:
The few additional computation would not cause any harm to performance. It just increases few computations in the compute* methods. Just for sanity verified a program with ListView of 10000000. ListView gets scrolled just as smooth.
Approving.